suite-native/module-send/src/components/SendForm.tsx

```diff
@@ -48,11 +62,45 @@ export const SendForm = ({ accountKey }: SendFormProps) => {
 
     const {
         handleSubmit,
-        formState: { isValid },
+        watch,
+        formState: { isValid: isFormValid },
     } = form;
+    const rawValues = watch();
+
+    useEffect(() => {
+        // debounce the redux action so it is not triggered on every keystroke
+        debounce(async () => {
+            if (isFormValid) {
+                // TODO: this object will be filled with more values in the future as the send form inputs will get more complex.
+                const formValues: FormState = {
+                    outputs: [
+                        {
+                            type: 'payment',
+                            address: rawValues.address,
+                            amount: rawValues.amount,
+                            token: null,
+                            fiat: '0',
+                            currency: { label: 'usd', value: '1000' },
+                        },
+                    ],
+                    isCoinControlEnabled: false,
+                    hasCoinControlBeenOpened: false,
+                    selectedUtxos: [],
+                    feeLimit: '0',
+                    feePerUnit: '0',
+                    options: [],
+                };
+
+                return await dispatch(
+                    sendFormActions.storeDraft({ accountKey, formState: formValues }),
+                );
+            }
+        });
+    }, [isFormValid, dispatch, debounce, accountKey, rawValues]);
 
-    const handleValidateForm = handleSubmit(() => {
-        // TODO: start on-device inputs validation via TrezorConnect
+    const handleNavigateToReview = handleSubmit(() => {
+        // Redirect to review screen where is on-device inputs validation via TrezorConnect started.
```

Comment:
nit: Are these comments really useful in your opinion? To me it just seems like clutter

Reply:
removed dd1094ff2a